### PR TITLE
Add OTEL to MDC for instrumentation using PiG/PNC subcommands

### PR DESCRIPTION
**Pull request by rnc**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/bacon/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -86,7 +86,7 @@
         <picocli.version>4.6.2</picocli.version>
         <pme.version>4.6</pme.version>
 
-        <pnc.version>2.4.1</pnc.version>
+        <pnc.version>2.5.0-SNAPSHOT</pnc.version>
```
Comment (rnc) [inline pom.xml:89]: Note this changes the version of PNC ... I can revert that if needed but for the instrumentation to function we need this.

**Review by vibe13 — APPROVED** [review]

File: cli/src/main/java/org/jboss/pnc/bacon/cli/App.java [hunk 2]
```diff
@@ -258,11 +262,15 @@ private void init(String command) {
...
-                    OtelCLIHelper.defaultSpanProcessor(OtelCLIHelper.defaultSpanExporter(endpoint)));
+            log.debug("Enabling OpenTelemetry collection on {} with service name {}", endpoint, service);
+            OTelCLIHelper.startOTel(
+                service,
+                command,
+                OTelCLIHelper.defaultSpanProcessor( OTelCLIHelper.defaultSpanExporter( endpoint)));
+            SpanContext current = Span.current().getSpanContext();
+            MDCUtils.addMDCFromOtelHeadersWithFallback( null, current, true );
+            // Temporary until MdcToHeadersFilter also handles the traceparent header.
+            OtelUtils.createTraceParentHeader( current ).forEach( MDC::put );
```
Comment (rnc) [inline cli/src/main/java/org/jboss/pnc/bacon/cli/App.java:273]: When https://github.com/project-ncl/pnc/pull/3654/ is merged this can be removed

**Comment by rnc** [comment]
retest this please

File: pom.xml (hunk 1 above)
Comment (thescouser89) [reply pom.xml:87]: Yup it's fine!

**Review by thescouser89 — APPROVED** [review]

**Pull request merged**